mmagic/models/editors/animatediff/attention_3d.py

```diff
@@ -0,0 +1,700 @@
+            #     query_dim=dim,
+            #     heads=num_attention_heads,
+            #     dim_head=attention_head_dim,
+            #     dropout=dropout,
+            #     bias=attention_bias,
+            #     cross_attention_dim=cross_attention_dim
+            #     if only_cross_attention else None,
+            #     upcast_attention=upcast_attention,
+            # )
+        else:
+            # TODO: Check whether replace this with models.editors
+            # .ddpm.attention.CrossAttention
+            self.attn1 = CrossAttention(
+                query_dim=dim,
+                heads=num_attention_heads,
+                dim_head=attention_head_dim,
+                dropout=dropout,
+                bias=attention_bias,
+                upcast_attention=upcast_attention,
+            )
+        self.norm1 = AdaLayerNorm(
+            dim, num_embeds_ada_norm
+        ) if self.use_ada_layer_norm else nn.LayerNorm(dim)
+
+        # Cross-Attn
+        if cross_attention_dim is not None:
+            self.attn2 = CrossAttention(
+                query_dim=dim,
+                cross_attention_dim=cross_attention_dim,
+                heads=num_attention_heads,
+                dim_head=attention_head_dim,
+                dropout=dropout,
+                bias=attention_bias,
+                upcast_attention=upcast_attention,
+            )
+        else:
+            self.attn2 = None
+
+        if cross_attention_dim is not None:
+            self.norm2 = AdaLayerNorm(
+                dim, num_embeds_ada_norm
+            ) if self.use_ada_layer_norm else nn.LayerNorm(dim)
+        else:
+            self.norm2 = None
+
+        # Feed-forward
+        self.ff = FeedForward(
+            dim, dropout=dropout, activation_fn=activation_fn)
+        self.norm3 = nn.LayerNorm(dim)
+
+        # Temp-Attn
+        assert unet_use_temporal_attention is not None
+        if unet_use_temporal_attention:
+            self.attn_temp = CrossAttention(
+                query_dim=dim,
+                heads=num_attention_heads,
+                dim_head=attention_head_dim,
+                dropout=dropout,
+                bias=attention_bias,
+                upcast_attention=upcast_attention,
+            )
+            nn.init.zeros_(self.attn_temp.to_out[0].weight.data)
+            self.norm_temp = AdaLayerNorm(
+                dim, num_embeds_ada_norm
+            ) if self.use_ada_layer_norm else nn.LayerNorm(dim)
+
+    def forward(self,
+                hidden_states,
+                encoder_hidden_states=None,
+                timestep=None,
+                attention_mask=None,
+                video_length=None):
+        """forward with hidden states, context and timestep."""
+        # SparseCausal-Attention
+        norm_hidden_states = (
+            self.norm1(hidden_states, timestep)
+            if self.use_ada_layer_norm else self.norm1(hidden_states))
+
+        # pdb.set_trace()
```

Comment:
some unused comments like this should be deleted.

Reply:
I've deleted unused comments.